ietf/doc/views_search.py

```diff
@@ -307,6 +307,32 @@ def cached_redirect(cache_key, url):
 
     return cached_redirect(cache_key, urlreverse('ietf.doc.views_search.search') + search_args)
 
+def ad_dashboard_group_type(doc):
+    # Return group type for document for dashboard.
+    # If doc is not defined return list of all possible
+    # group types
+    if not doc:
+        return ('I-D', 'RFC', 'Conflict Review', 'Status Change', 'Charter')
+    if doc.type.slug=='draft':
+        if doc.get_state_slug('draft') == 'rfc':
+            return 'RFC'
+        elif doc.get_state_slug('draft') == 'active' and doc.get_state_slug('draft-iesg') and doc.get_state('draft-iesg').name in ('RFC Ed Queue Internet-Draft'):
+            return 'RFC'
+        elif doc.get_state_slug('draft') == 'active' and doc.get_state_slug('draft-iesg') and doc.get_state('draft-iesg').name in ('Dead', 'I-D Exists', 'AD is watching'):
+             return None
+        elif doc.get_state('draft').name in ('Expired', 'Replaced'):
+            return None
+        else:
+            return 'I-D'
+    elif doc.type.slug=='conflrev':
```

Comment:
Similar to @rjsparks's comment above would be better to make use of the `DocTypeName` data rather than repeating it here. A single `doc.type.slug in []` test could replace a bunch of these cases